Formula/u/unciv.rb

```diff
@@ -0,0 +1,24 @@
+class Unciv < Formula
+  desc "Open-source Android/Desktop remake of Civ V"
+  homepage "https://github.com/yairm210/Unciv"
+  url "https://github.com/yairm210/Unciv/releases/download/4.15.14/Unciv.jar"
+  sha256 "23ba482998b212579f53f677e014fc12cd7b9fa8fe3ffb6c4dbe9a03fcf1be6c"
+  license "MPL-2.0"
+
+  livecheck do
+    url :stable
+    regex(/^(\d+\.\d+\.\d+(-patch\d+)?)$/i)
```

Comment:
I dont think we are able to catch `-patch1` cc @samford 

any reason not using semver?

Reply:
I checked with brew livecheck, and saw that it could catch all the versions
As for semver... It's a historic reason, and not a particularly good one tbh 😕

Reply:
```suggestion
    regex(/^v?(\d+(?:\.\d+)+(?:[._-]?patch\d*)?)$/i)
```

A few tweaks to make this a little more flexible (e.g., there's at least one tag that uses a `-patch` suffix with no digit) and use a non-capturing group for the suffix.

It's worth mentioning that this will only match tags like `1.2.3`, `1.2.3-patch1`, etc. There appear to be 50 different version suffixes that don't start with `patch` (e.g., `-libgdx-1.13.1`, `-apk-minify`, `-redo`, `-gp`, `-msi-test-7`, `-cli3`, `-macOS`, etc.), so you will have to make sure to strictly adhere to the aforementioned tag format for any versions that we should use. For example, this regex doesn't match `4.15.6-patch1-macOS` because of the trailing `-macOS`.

Reply:
Ah, if it makes sense for your release process, it would be helpful if you set test releases on GitHub as pre-release. Currently, releases like `4.15.1-libgdx-1.13.1` are treated the same as stable versions like `4.15.1`, `4.15.11-patch1`, etc., so it's not possible to reliably identify the newest stable version from the "latest" release (https://github.com/yairm210/Unciv/releases/latest). It's a stable version at the moment but test releases can also become "latest" if they're not set as pre-release (e.g., https://web.archive.org/web/20250101082032/https://github.com/yairm210/Unciv/releases).

This isn't an issue for us yet as livecheck is checking the Git tags here but we may expand usage of the `GithubLatest` strategy in the future (after working through some technical limitations on our end) and it would be a problem then. Making adjustments to your release workflow to ensure that "latest" is always a stable version would be helpful for us but also anyone else who is checking releases (e.g., it would make it clear to users which releases are stable).

Reply:
I remember reading in your docs that you warn against having too many pre-releases because they can fill the "first page" and then the liveupdate on github can't find the latest "real version" 🤔 
But you're right that it does make sense for us to do this, and it should probably be an easy change... I'll meditate on this

Reply:
I remembered that I already did this, and turns out this was actually a minor error in our CI!
It used to be that these 'test releases' would not contain the version they were based on and then everything worked as expected
But we're long since standardized that the test releases do contain the version, which meant that the regex for determining validity was always triggered... 😅 

[Fixed here](https://github.com/yairm210/Unciv/commit/5ee160f84f7d49b5138819195d966a4e8bc86513)
Should work correctly now, good catch and thanks for the help!